dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/CompanyDataController.kt

```diff
@@ -47,19 +54,49 @@ class CompanyDataController(
         )
         return ResponseEntity.ok(
             companyManager.searchCompaniesAndGetApiModel(
-                CompanySearchFilter(
-                    searchString ?: "",
-                    onlyCompanyNames,
-                    dataTypes ?: setOf(),
-                    countryCodes ?: setOf(),
-                    sectors ?: setOf(),
-                    onlyCurrentUserAsUploader,
+                StoredCompanySearchFilter(
+                    searchString = searchString ?: "",
+                    nameOnlyFilter = onlyCompanyNames,
+                    dataTypeFilter = dataTypes.map { it.name },
```

Comment:
This will need a default too after we make dataTypes a nullable field again, right?

Reply:
Yup, done.